src/System.Net.Sockets/src/System.Net.Sockets.csproj

```diff
@@ -223,9 +213,178 @@
+      <Link>Interop\Unix\libc\Interop.recv.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.recvfrom.cs">
+      <Link>Interop\Unix\libc\Interop.recvfrom.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.recvmsg.cs">
+      <Link>Interop\Unix\libc\Interop.recvmsg.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.select.cs">
+      <Link>Interop\Unix\libc\Interop.select.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.send.cs">
+      <Link>Interop\Unix\libc\Interop.send.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.sendmsg.cs">
+      <Link>Interop\Unix\libc\Interop.sendmsg.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.sendto.cs">
+      <Link>Interop\Unix\libc\Interop.sendto.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.setsockopt.cs">
+      <Link>Interop\Unix\libc\Interop.setsockopt.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.shutdown.cs">
+      <Link>Interop\Unix\libc\Interop.shutdown.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.socket.cs">
+      <Link>Interop\Unix\libc\Interop.socket.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.timeval.cs">
+      <Link>Interop\Unix\libc\Interop.timeval.cs</Link>
+    </Compile>
+
+    <Compile Include="$(CommonPath)\Interop\Unix\System.Native\Interop.Close.cs">
+      <Link>Interop\Unix\System.Native\Interop.Close.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Unix\System.Native\Interop.Fcntl.cs">
+      <Link>Interop\Unix\System.Native\Interop.Fcntl.cs</Link>
+    </Compile>
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(TargetsLinux)' == 'true' ">
+    <Compile Include="System\Net\Sockets\SocketAsyncContext.cs" />
+    <Compile Include="System\Net\Sockets\SocketAsyncEngine.cs" />
+
+    <Compile Include="$(CommonPath)\Interop\Linux\Interop.Libraries.cs">
+      <Link>Interop\Linux\Interop.Libraries.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Linux\libc\Interop.epoll_create1.cs">
+      <Link>Interop\Linux\libc\Interop.epoll_create1.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Linux\libc\Interop.epoll_ctl.cs">
+      <Link>Interop\Linux\libc\Interop.epoll_ctl.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Linux\libc\Interop.epoll_event.cs">
+      <Link>Interop\Linux\libc\Interop.epoll_event.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\Linux\libc\Interop.epoll_wait.cs">
+      <Link>Interop\Linux\libc\Interop.epoll_wait.cs</Link>
+    </Compile>
+
+    <Compile Include="$(CommonPath)\Interop\Unix\libc\Interop.addrinfo.cs">
```

Comment:
These files are marked as "Unix". Why not have the same conditional on the ItemGroup as in System.Net.Primitives.csproj. (Same question applies for NameResolution.)